src/lib/db/index.ts

```diff
@@ -10,31 +10,40 @@ import { UUID } from '../../models/uuid'
 
 const MODELS_DIRECTORY = path.join(__dirname, '../../models')
 
+export interface Models<V> {
+  [key: string | number]: V[keyof V]
+}
+
+export type Query = Knex.QueryBuilder
+
 /** Creates a connection to the postgres instance
- * usage: var db = new Database().init()
+ * usage: var client = new Database()
+ * db = client.init()
  * db.Example().where(id); db.Table.select(id);
  */
 export default class Database {
   public client: Knex
   private log: Logger
-  public db: any
+  public db: () => Models<Query> | any
 
   constructor() {
     this.log = logger
     this.client = knex(pgConfig)
-    this.db = (models: any = {}) => {
-      this.log.debug('initializing db models')
-      fs.readdirSync(MODELS_DIRECTORY).forEach((file: string): any => {
+    this.db = (models: Models<Query> = {}) => {
+      fs.readdirSync(MODELS_DIRECTORY).forEach((file: string) => {
         const { name } = path.parse(file)
         // TODO check if table exists -> append to the db object
-        if (name != 'index.d') models[name] = () => this.client(name)
+        if (name != 'index.d' && name != 'health') {
+          this.log.debug(`initializing ${name} db model`)
+          models[name] = () => this.client(name)
+        }
```

Comment:
This is pretty jank. It relies on the fact that no path transformations take place and if I'm right the only purpose is to get the name of the table? I'm not sure I've got a better solution right now but even a list of strings at the top of this file might be better with a runtime dynamic `import` call to ensure the file exists? At least that way if something like webpack were ever to run here it would see the dynamic import and know not to transform the names

Reply:
Is this happening for this PR or not?

Reply:
No